webserver.cpp

```diff
@@ -764,7 +773,7 @@ int http_parse_multipart_body(struct webserver_t *client, unsigned char *buf, ui
   uint16_t hasread = MIN(WEBSERVER_BUFFER_SIZE-client->ptr, len);
   uint16_t rpos = 0, loop = 1;
 
-  while(rpos < len) {
+  while((rpos < len) || ((loop == 1) && (client->ptr > 0))) { //do always first run with initial loop==1 because read buffer can have been already moved completly into client buffer
```

Comment:
The inline comment already explains it a bit. The first call of this function can be called with a len=0 as all data is already in the (large) client buffer but it still needs to be parsed as a multipart. As loop is initialized by 1 and it is checked that the client buffer does indeed containt data, the while run will start and enter the next while loop=1 loop.